### PR TITLE
Fixed broken diffusive routing in Lower Colorado example

### PR DESCRIPTION
**Pull request by JurgenZach-NOAA**
Recent updates caused the diffusive routing for NHD in the Lower Colorado example to no longer worked.  

## Additions

- Added new Routelink file (2.1), subset w.r.t. the diffusive study area including Lower Colorado and other river systems in the Gulf Coast and New England
- Added updated qlat files (and subset the same way)
- Added Lakeparm file (2.1)

## Removals

- Removed old Routelink- and Lakeparm files
- Removed old qlat forcing files

## Changes

- Turned on diffusive routing in test_AnA.yaml by default

## Testing

1. In the test/LowerColorado_TX directory, issue: >> python -m nwm_routing -f -V3 test_AnA.yaml

## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Windows
- [ ] Linux
- [ ] Browser

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

**Comment by AminTorabi-NOAA** [comment]
@JurgenZach-NOAA If you uncomment `output_parameters:` and `stream_output` to save the results. It will throw an error that `AttributeError: 'str' object has no attribute 'keys'`. Can you fix the error

**Comment by JurgenZach-NOAA** [comment]
> @JurgenZach-NOAA If you uncomment `output_parameters:` and `stream_output` to save the results. It will throw an error that `AttributeError: 'str' object has no attribute 'keys'`. Can you fix the error

This issue was not actually related to the PR - however, it's corrected now. What happened was that, in cases without assimilated gages and NHD networks, link_lake_crosswalk was set to "None" instead of (Boolean) False. So, the output routine thought there's a link_lake_crosswalk when there couldn't be. Corrected it.

**Comment by JurgenZach-NOAA** [comment]
Also turned on the output in the example - there should always be an output!

File: src/troute-nwm/src/nwm_routing/output.py [hunk 1]
```diff
@@ -174,7 +174,7 @@ def nwm_output_generator(
             d_df = flow_df.iloc[:,2::3]
 
         # replace waterbody lake_ids with outlet link ids
-        if link_lake_crosswalk:
+        if (link_lake_crosswalk and not link_lake_crosswalk=='NONE'):
```
Comment (AminTorabi-NOAA) [inline src/troute-nwm/src/nwm_routing/output.py:177]: @JurgenZach-NOAA  `link_lake_crosswalk` shouldn't be "NONE", its a dictionary, it's either empty or have values in that. The fact that it become 'NONE" shows something is wrong. If you look at where this function being called in `__main__ .py` (line 1791) and put a trace before the function you'll see `link_lake_crosswalk` has a value, but right after the function being called it become "NONE". The reason is one input (`duplicate_ids_df`) is missing from the function when it being called

**Comment by JurgenZach-NOAA** [comment]
image

nwm_out_generator in the V3 main branch had a missing argument, the duplicate_ids_df. It is only relevant for HYfeatures, so it can be set to an empty df here, but still needs to be provided.

**Review by AminTorabi-NOAA — APPROVED** [review]

**Pull request merged**